### PR TITLE
Added LA-Workspace bicep

### PR DESCRIPTION
**Pull request by MrMCake**
# Change

- Added LA-Workspace bicep

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: arm/Microsoft.OperationalInsights/workspaces/deploy.bicep [hunk 1]
```diff
@@ -0,0 +1,844 @@
...
+
+resource logAnalyticsWorkspace_lock 'Microsoft.Authorization/locks@2016-09-01' = if (lockForDeletion) {
+  name: '${logAnalyticsWorkspaceName}-LADoNotDelete'
+  properties: {
+    level: 'CanNotDelete'
+  }
+  scope: logAnalyticsWorkspace
+}
+
+module rbac_name './.bicep/nested_rbac.bicep' = [for (item, i) in roleAssignments: {
```
Comment (eriqua) [inline arm/Microsoft.OperationalInsights/workspaces/deploy.bicep:829]: I'd suggest `logAnalyticsWorkspace_rbac`

Comment (eriqua) [reply arm/Microsoft.OperationalInsights/workspaces/deploy.bicep:829]: I'd suggest `[for (roleAssignment, index) in roleAssignments: {`

File: arm/Microsoft.OperationalInsights/workspaces/deploy.bicep [hunk 2]
```diff
@@ -0,0 +1,844 @@
...
+resource linkedServices 'linkedServices@2020-03-01-preview' = if (!empty(automationAccountId)) {
+  name: 'Automation'
+  properties: {
+    resourceId: automationAccountId
+  }
+}
+}
+
+@batchSize(1) // Serial loop deployment
+resource solutionRes 'Microsoft.OperationsManagement/solutions@2015-11-01-preview' = [for solution in solutions: if (!empty(solutions)) {
```
Comment (eriqua) [inline arm/Microsoft.OperationalInsights/workspaces/deploy.bicep:804]: I'd suggest `solutions`, unless creating conflicts with the solutions parameter

File: arm/Microsoft.OperationalInsights/workspaces/deploy.bicep [hunk 3]
```diff
@@ -0,0 +1,844 @@
...
+resource logAnalyticsWorkspace_datasources_AzureActivityLog 'datasources@2020-03-01-preview' = [for item in activityLogAdditionalSubscriptionIDs: {
+  kind: 'AzureActivityLog'
+  name: '${(empty(activityLogAdditionalSubscriptionIDs) ? 'placeholder' : item)}'
+  properties: {
+    linkedResourceId: '/subscriptions/${item}/providers/microsoft.insights/eventTypes/management'
+  }
+  dependsOn: [
+    logAnalyticsWorkspace
+  ]
+}]
```
Comment (eriqua) [inline arm/Microsoft.OperationalInsights/workspaces/deploy.bicep:776]: I'd vote for parameterizing this as much as possible and have a module iterating on datasourcers resources

**Review by eriqua — CHANGES_REQUESTED** [review]

File: arm/Microsoft.OperationalInsights/workspaces/deploy.bicep (hunk 2 above)
Comment (MrMCake) [reply arm/Microsoft.OperationalInsights/workspaces/deploy.bicep:804]: It does. That's the only reason I used it. I guess the alternative would be the rename the parameter to e.g. `solutionsDef` os similar

Comment (MrMCake) [reply arm/Microsoft.OperationalInsights/workspaces/deploy.bicep]: updated the parameter

File: arm/Microsoft.OperationalInsights/workspaces/deploy.bicep (hunk 3 above)
Comment (MrMCake) [reply arm/Microsoft.OperationalInsights/workspaces/deploy.bicep:776]: Agreed. But let's do that in a separate PR as this seems to be a bigger change (to me)

File: arm/Microsoft.OperationalInsights/workspaces/deploy.bicep (hunk 1 above)
Comment (MrMCake) [reply arm/Microsoft.OperationalInsights/workspaces/deploy.bicep]: done

**Review by eriqua — APPROVED** [review]
good to go 🚀

**Pull request merged**